src/OpenApi/test/Transformers/Implementations/OpenApiSchemaReferenceTransformerTests.cs

```diff
@@ -289,4 +289,58 @@ public async Task TypeModifiedWithSchemaTransformerMapsToDifferentReferenceId()
             Assert.False(responseSchema.GetEffective(document).Extensions.TryGetValue("x-my-extension", out var _));
         });
     }
+
+    [Fact]
+    public static async Task ProducesStableSchemaRefsForListOf()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapPost("/api", (List<Todo> todo) => { });
+        builder.MapPost("/api-2", (List<Todo> todo) => { });
+
+        // Assert -- call twice to ensure the schema reference is stable
+        await VerifyOpenApiDocument(builder, VerifyDocument);
+        await VerifyOpenApiDocument(builder, VerifyDocument);
+
+        static void VerifyDocument(OpenApiDocument document)
+        {
+            var operation = document.Paths["/api"].Operations[OperationType.Post];
+            var requestBody = operation.RequestBody.Content["application/json"];
+            var requestBodySchema = requestBody.Schema;
+
+            var operation2 = document.Paths["/api-2"].Operations[OperationType.Post];
+            var requestBody2 = operation2.RequestBody.Content["application/json"];
+            var requestBodySchema2 = requestBody2.Schema;
+
+            // {
+            //   "$ref": "#/components/schemas/TodoList"
+            // }
+            // {
+            //   "$ref": "#/components/schemas/TodoList"
+            // }
+            // {
+            //   "components": {
+            //     "schemas": {
+            //       "ArrayOfTodo": {
+            //         "type": "array",
+            //         "items": {
+            //           "$ref": "#/components/schemas/Todo"
+            //         }
+            //       }
+            //     }
+            //   }
+            // }
+
+            // Both list types should point to the same reference ID
+            Assert.Equal("ArrayOfTodo", requestBodySchema.Reference.Id);
```

Comment:
To test my understanding, this would have been `ArrayOfTodo2` in the buggy case?